src/uu/test/src/test.rs

```diff
@@ -97,9 +97,17 @@ fn eval(stack: &mut Vec<Symbol>) -> ParseResult<bool> {
             Ok(!result)
         }
         Some(Symbol::Op(Operator::String(op))) => {
-            let b = stack.pop();
-            let a = stack.pop();
-            Ok(if op == "!=" { a != b } else { a == b })
+            let b = pop_literal!();
+            let a = pop_literal!();
+            if op == "!=" {
+                Ok(a != b)
+            } else if op == "<" {
+                Ok(a < b)
+            } else if op == ">" {
+                Ok(a > b)
+            } else {
+                Ok(a == b)
+            }
```

Comment:
This looks like a case for a `match`, something like:
```suggestion
            match op.to_string_lossy().as_ref() {
                "!=" => Ok(a != b),
                "<" => Ok(a < b),
                ">" => Ok(a > b),
                _ => Ok(a == b),
            }
```